packages/integrations/components/pico/utils.js

```diff
@@ -0,0 +1,33 @@
+export default function mountPicoNodes(pageInfo) {
+  let nodesMounted = false;
+
+  // Mount the Signal and Rule nodes to be targeted by the Pico embed.
+  const documentBody = document.getElementsByTagName('body')[0];
+  // Append a container node for the Pico signals to target into the DOM if one doesn't already exist.
+  if (documentBody && ! document.getElementById('PicoSignal-container')) {
+    // Create the signal node.
+    const signalNode = document.createElement('div');
+    // Set the attributes.
+    signalNode.setAttribute('id', 'PicoSignal-container');
+    signalNode.setAttribute('class', 'PicoSignal');
+    signalNode.setAttribute('style', 'display: none');
+    // Append the node into the DOM.
+    documentBody.appendChild(signalNode);
```

Comment:
Is this maybe an excuse to use [portals](https://reactjs.org/docs/portals.html#usage)?  So like:
```javascript
const SignalNode = () => (
  <div
    id="PicoSignal-container"
    class="PicoSignal"
    style={{ display: 'none' }}
  />
);

React.createPortal(SignalNode, document.getElementById('PicoSignal-container'));
```

Reply:
@ostowe Great idea. I tried implementing that but could not get the nodes to mount in a reliable way for all renders. I think since these components live outside of the context of React and are only used for interacting/tracking actions in the Pico widget embed that the plain JS nodes should be sufficient here.